packages/bot-cleaning/src/MarketCleaner.ts

```diff
@@ -149,27 +152,37 @@ export class MarketCleaner {
         this.#whitelistedDustCleaningMaker &&
         this.#whitelistedDustCleaningMaker.has(offer.maker.toLowerCase())
       ) {
+        if (!offer.price) {
+          continue;
+        }
+
         const inboundTkn =
           ba == "bids" ? semibook.market.base : semibook.market.quote;
         const outboundTkn =
           ba == "bids" ? semibook.market.quote : semibook.market.base;
 
-        let takerWants = new Big(1).div(new Big(10).pow(outboundTkn.decimals));
-        let takerGives = inboundTkn.fromUnits(
-          takerWants.mul(offer.price!).toString()
-        );
-        if (outboundTkn.decimals > inboundTkn.decimals) {
-          takerGives = new Big(1).div(new Big(10).pow(inboundTkn.decimals));
-          takerWants = takerGives.mul(new Big(1).div(offer.price!).round());
+        let price = offer.price;
+        if (ba === "bids") {
+          price = new Big(1).div(offer.price);
+        }
+
+        /* this logic find the minimum takerGives */
```

Comment:
"Minimum" in what sense?

Reply:
Added a big comment in the strategies.ts file